core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -276,7 +285,15 @@ private boolean usePendingAcquisition(SessionProtocol desiredProtocol, PoolKey k
             if (cause == null) {
                 final SessionProtocol actualProtocol = pch.protocol();
                 if (actualProtocol.isMultiplex()) {
-                    promise.complete(pch);
+                    final HttpSession session = HttpSession.get(pch.get());
+                    logger.info("session.unfinishedResponses(): {}, session.maxUnfinishedResponses(): {}",
+                                session.unfinishedResponses(), session.maxUnfinishedResponses());
+                    // unfinishedResponses >= maxUnfinishedResponses - current connecting request
+                    if (session.unfinishedResponses() >= session.maxUnfinishedResponses() - 1) {
```

Comment:
Better extracting them into a local variable if you intend to keep the logging above.

Reply:
added a local variable and removed the comment -- let me know if you prefer I leave the comment though 🙇